### PR TITLE
deposit api: Config api headers 

### PR DESCRIPTION
**Pull request by philippgualdi**
The implementation allows you to configure the BASE_HEADERS for other modules using the DepositFormApp and the REST
api client.

### CODE REVIEW

File: src/lib/DepositApiClient.js [hunk 1]
```diff
@@ -111,8 +103,8 @@ export class RDMDepositApiClient extends DepositApiClient {
   async createDraft(draft) {
     const payload = this.recordSerializer.serialize(draft);
     return this._createResponse(() =>
-      axiosWithConfig.post(this.createDraftURL, payload, {
-        headers: BASE_HEADERS["vnd+json"],
+      this.axiosWithConfig.post(this.createDraftURL, payload, {
+        headers: this.apiHeaders["vnd+json"],
```
Comment (kpsherva) [inline src/lib/DepositApiClient.js:107]: with BASE_HEADERS removed, we cannot assure `vnd+json` to be present in the apiHeaders object

Comment (philippgualdi) [reply src/lib/DepositApiClient.js:107]: The BASE_HEADERS are set in the configuration in the app inveniosoftware/invenio-app-rdm#1917.

Comment (philippgualdi) [reply src/lib/DepositApiClient.js:107]: The BASE_HEADERS will be set in the invenio_app_rdm and allows other modules to set their own headers.

Comment (kpsherva) [reply src/lib/DepositApiClient.js:107]: I understand, however this solution relies on defining the headers in a repository which depends on react-invenio-deposit. As a result there is a "mutual dependency" created - module cannot be used without invenio-app-rdm. 
If the assumption of having `vnd+json` header in this.apiHeaders object is not fulfiled then headers will be evaluated as `undefined`

Comment (philippgualdi) [reply src/lib/DepositApiClient.js:118]: I see your point and pushed a solution where no mutual dependency is created.

File: src/lib/DepositApiClient.js [hunk 2]
```diff
@@ -36,10 +28,21 @@ export class DepositApiClientResponse {
...
   /* eslint-disable no-unused-vars */
-  constructor(createDraftURL, recordSerializer) {
+  constructor(apiHeaders, createDraftURL, recordSerializer) {
     if (this.constructor === DepositApiClient) {
       throw new Error("Abstract");
     }
+
+    this.apiHeaders = Object.assign({}, BASE_HEADERS, apiHeaders);
+
+    this.apiConfig = {
```
Comment (ntarocco) [inline src/lib/DepositApiClient.js:38]: Suggestion: make the entire config overridable, something like this (pseudocode):
```
constructor(extraApiConfig, createDraftURL, recordSerializer) {
...
}

const extraHeaders = _get(extraApiConfig, "headers", {});
this.apiHeaders = Object.assign({}, BASE_HEADERS, extraHeaders);

this.apiConfig = {
    withCredentials: true,
    ...extraApiConfig,
}
```

File: src/lib/DepositFormApp.js [hunk 3]
```diff
@@ -38,9 +38,11 @@ export class DepositFormApp extends Component {
           props.config.custom_fields.vocabularies
         );
 
+    const apiHeaders = props.config.apiHeaders ? props.config.apiHeaders : null;
+
     const apiClient = props.apiClient
       ? props.apiClient
-      : new RDMDepositApiClient(props.config.createUrl, recordSerializer);
+      : new RDMDepositApiClient(apiHeaders, props.config.createUrl, recordSerializer);
```
Comment (ntarocco) [inline src/lib/DepositFormApp.js:45]:
```suggestion
      : new RDMDepositApiClient({headers: apiHeaders}, props.config.createUrl, recordSerializer);
```

**Review by ntarocco — APPROVED** [review]
LGTM! Thank you!
@kpsherva @zzacharo if you are happy with this, we can merge and release!
@philippgualdi do you need this backported it to any release?

**Comment by philippgualdi** [comment]
> LGTM! Thank you! @kpsherva @zzacharo if you are happy with this, we can merge and release! @philippgualdi do you need this backported it to any release?

Backported to an older release is not necessary. I appreciate your quick response, thank you! :)

**Review by kpsherva — APPROVED** [review]

**Review by zzacharo — APPROVED** [review]

**Pull request merged**